crates/notion-core/src/tool.rs

```diff
@@ -348,3 +355,94 @@ impl Tool for Yarn {
         }
     }
 }
+
+impl Tool for Npm {
+    fn new(session: &mut Session) -> Fallible<Self> {
+        session.add_event_start(ActivityKind::Npm);
+
+        let mut args = args_os();
+        let exe = arg0(&mut args)?;
+        if let Some(ref platform) = session.current_platform()? {
+            session.prepare_image(platform)?;
+
+            if platform.yarn.is_some() {
+                // If Yarn is enabled for the current project, warn the user that they shouldn't be using npm
+                style::display_warning(&String::from("This project is configured to use yarn, npm should be avoided"));
+            }
+            Ok(Self::from_components(&exe, args, &platform.path()?))
+        } else {
+            // Using 'Node' as the tool name since the npm version is derived from the Node version
+            // This way the error message will prompt the user to add 'Node' to their toolchain, instead of 'npm'
+            throw!(NoSuchToolError {
```

Comment:
I'm not 100% confident this is the right UX but we don't need to change it for now -- we can ship this and play with it.